src/main/java/bio/terra/cli/service/utils/HttpUtils.java

```diff
@@ -108,4 +118,78 @@ public static HttpResponse sendHttpRequest(
     // return a POJO that includes both the response body and status code
     return new HttpResponse(responseBody.toString(), statusCode);
   }
+
+  /**
+   * Helper method to do retries. Uses {@link #DEFAULT_MAXIMUM_RETRIES} for maximum number of
+   * retries and {@link #DEFAULT_MILLISECONDS_SLEEP_FOR_RETRY} for the time to sleep between
+   * retries.
+   *
+   * @param makeRequest function to perform the request
+   * @param isRetryable function to test whether the exception is retryable or not
+   * @param <T> type of the Http response (i.e. return type of the makeRequest function)
+   * @return the Http response
+   * @throws Exception if makeRequest throws an exception that is not retryable, or if the maximum
+   *     number of retries was exhausted
+   */
+  public static <T, E extends Exception> T callWithRetries(
+      HttpRequestOperator<T, E> makeRequest, Predicate<Exception> isRetryable)
+      throws E, InterruptedException {
+    return callWithRetries(
+        DEFAULT_MAXIMUM_RETRIES, DEFAULT_MILLISECONDS_SLEEP_FOR_RETRY, makeRequest, isRetryable);
+  }
+  /**
+   * Helper method to do retries.
+   *
+   * @param maxRetries maximum number of times to retry
+   * @param sleepDurationMS number of milliseconds to sleep between tries
+   * @param makeRequest function to perform the request
+   * @param isRetryable function to test whether the exception is retryable or not
+   * @param <T> type of the Http response (i.e. return type of the makeRequest function)
+   * @return the Http response
+   * @throws Exception if makeRequest throws an exception that is not retryable, or if the maximum
+   *     number of retries was exhausted
+   */
+  public static <T, E extends Exception> T callWithRetries(
+      int maxRetries,
+      int sleepDurationMS,
+      HttpRequestOperator<T, E> makeRequest,
+      Predicate<Exception> isRetryable)
+      throws E, InterruptedException {
```

Comment:
I don't see this throwing an InterruptedException. I don't think we even want our callers to care about InterruptedExceptions. Throws `E` is enough even if we run out of retries.

Reply:
Thread.sleep throws an InterruptedException. I suppose I could put a try/catch in the retry method and wrap it in a runtime exception, but I think it's a more informative error message in the calling methods.